### DISCUSSION TITLE
Is there a workaround to show QR code in svg for Gmail?

### DISCUSSION THREAD

**Discussion by holahoon** [zpao/qrcode.react#354] [General]
It seems like Gmail still doesn't support svg format in their email 🤦🏻
I've been trying to include the generated QR code inside users' emails, but it's Gmail that doesn't display svg images.
Been looking around to see if anybody had the same situation, but guess not?
Does anybody know if there's a way to show the QR code in Gmail?

**Comment by zpao** [comment]
I assume you're using some React email renderer which generates the markup for emails...

Eh, because the two rendering modes here are SVG and Canvas, I don't think either are good options. My recommendation would be using something that renders to PNG, which would really need to happen server-side. Or have an api type route on your server which renders qrcodes on request so then you could just have `<img src=>` in your markup.
